### PR TITLE
adding window setTimeout term

### PR DESCRIPTION
**Pull request by RaffyRod**
### Description

* Adding the Window  `setTimeout` term for Javascript.

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

**Comment by RaffyRod** [comment]
@yangc95  need help, CI is failing but I don't know why?

**Comment by yangc95** [comment]
@RaffyRod no worries! After running `yarn test` it looks like it was an issue with the metadata's description. Since it's enclosed in single quotes, no single quotes should exist within the string. I'll conduct the first review for this.

File: content/javascript/concepts/setTimeout/setTimeout.md
Comment (yangc95) [inline content/javascript/concepts/setTimeout/setTimeout.md:2]:
```suggestion
Title: 'setTimeout()'
```

Comment (yangc95) [reply content/javascript/concepts/setTimeout/setTimeout.md:2]: A little tricky, we'll have to wait to merge this PR until the window concept in issue [#2007](https://github.com/Codecademy/docs/issues/2007) is merged.

But we'll handle this when the time comes or when the mentioned issue's PR is merged you can feel free to do this yourself; this folder/file will go into a folder named `window/terms/`.

File: content/javascript/concepts/setTimeout/setTimeout.md
Comment (yangc95) [inline content/javascript/concepts/setTimeout/setTimeout.md:3]:
```suggestion
Description: 'Explain what the function does here.'
```

We'll model this after a terms entry still. Make sure to start off with a present-tense verb

File: content/javascript/concepts/setTimeout/setTimeout.md
Comment (yangc95) [inline content/javascript/concepts/setTimeout/setTimeout.md:16]:
```suggestion
`setTimeout()` is a function serviced globally by the window object provided by the user's browser, it is not part of JavaScript because it is not a JavaScript function.
```

We'll refer to the function as such throughout the rest of the entry as well. JavaScript is available on the browser, Node.js is what we use to write in JavaScript locally

File: content/javascript/concepts/setTimeout/setTimeout.md
Comment (yangc95) [inline content/javascript/concepts/setTimeout/setTimeout.md:25]: should function be referred to as callback instead?

File: content/javascript/concepts/setTimeout/setTimeout.md
Comment (yangc95) [inline content/javascript/concepts/setTimeout/setTimeout.md:24]:
```suggestion
```pseudo
```

File: content/javascript/concepts/setTimeout/setTimeout.md
Comment (yangc95) [inline content/javascript/concepts/setTimeout/setTimeout.md:66]:
```suggestion
```shell
```

File: content/javascript/concepts/setTimeout/setTimeout.md
Comment (yangc95) [inline content/javascript/concepts/setTimeout/setTimeout.md:58]:
```suggestion
```

File: content/javascript/concepts/setTimeout/setTimeout.md
Comment (yangc95) [inline content/javascript/concepts/setTimeout/setTimeout.md:32]:
```suggestion
## Examples
```

File: content/javascript/concepts/setTimeout/setTimeout.md
Comment (yangc95) [inline content/javascript/concepts/setTimeout/setTimeout.md:38]:
```suggestion
```

File: content/javascript/concepts/setTimeout/setTimeout.md
Comment (yangc95) [inline content/javascript/concepts/setTimeout/setTimeout.md:34]:
```suggestion
The following pieces of code will show some examples about the `setTimeout()` implementation:
```

File: content/javascript/concepts/setTimeout/setTimeout.md
Comment (yangc95) [inline content/javascript/concepts/setTimeout/setTimeout.md:44]:
```suggestion
```

File: content/javascript/concepts/setTimeout/setTimeout.md
Comment (yangc95) [inline content/javascript/concepts/setTimeout/setTimeout.md:48]:
```suggestion
The output from the `sayHello` function will be displayed after 4 seconds:
```

File: content/javascript/concepts/setTimeout/setTimeout.md
Comment (yangc95) [inline content/javascript/concepts/setTimeout/setTimeout.md:24]:
```suggestion
```pseudo
```

File: content/javascript/concepts/setTimeout/setTimeout.md
Comment (yangc95) [inline content/javascript/concepts/setTimeout/setTimeout.md:54]:
```suggestion
An example using [arrow functions](https://www.codecademy.com/resources/docs/javascript/arrow-functions) would be:
```

Comment (yangc95) [reply content/javascript/concepts/setTimeout/setTimeout.md:54]: Or you can change this line to a header like "Example Using Arrow Functions"

File: content/javascript/concepts/setTimeout/setTimeout.md
Comment (yangc95) [inline content/javascript/concepts/setTimeout/setTimeout.md:64]:
```suggestion
The output from the arrow function will be displayed after 2 seconds:
```

File: content/javascript/concepts/setTimeout/setTimeout.md
Comment (yangc95) [inline content/javascript/concepts/setTimeout/setTimeout.md:74]:
```suggestion
```

File: content/javascript/concepts/setTimeout/setTimeout.md
Comment (yangc95) [inline content/javascript/concepts/setTimeout/setTimeout.md:85]:
```suggestion
The output will be displayed after 3 seconds:
```

File: content/javascript/concepts/setTimeout/setTimeout.md
Comment (yangc95) [inline content/javascript/concepts/setTimeout/setTimeout.md:87]:
```suggestion
```shell
```

**Review by yangc95 — COMMENTED** [review]
@RaffyRod good examples! I've left some additional comments.

Tag me once you've made appropriate changes and if you need any assistance or have concerns!

**Comment by RaffyRod** [comment]
@yangc95 already made changes suggested, please check

File: content/javascript/concepts/setTimeout/setTimeout.md
Comment (yangc95) [reply content/javascript/concepts/setTimeout/setTimeout.md:34]: @RaffyRod style

File: content/javascript/concepts/setTimeout/setTimeout.md
Comment (yangc95) [reply content/javascript/concepts/setTimeout/setTimeout.md:38]: @RaffyRod remove space 👾

File: content/javascript/concepts/setTimeout/setTimeout.md
Comment (yangc95) [reply content/javascript/concepts/setTimeout/setTimeout.md:66]: output is not a valid code block. Please inquire for a list of available ones or check documentation

File: content/javascript/concepts/setTimeout/setTimeout.md
Comment (yangc95) [inline content/javascript/concepts/setTimeout/setTimeout.md:70]: Change to AP style title

Comment (yangc95) [reply content/javascript/concepts/setTimeout/setTimeout.md:70]: also rephrase to example using parameters. Consistency is most important so if you want to break down the headers to "Example using <whatever specification>", I'd recommend changing the rest to fit that same structure. Change these to h2's instead of h3 for each header where you'll include an example.

File: content/javascript/concepts/setTimeout/setTimeout.md
Comment (yangc95) [inline content/javascript/concepts/setTimeout/setTimeout.md:54]: This would be a header named example using arrow functions. In which case I recommend having 1-2 sentences explaining the code below. Within those sentences, you can include the arrow functions entry link.

File: content/javascript/concepts/setTimeout/setTimeout.md
Comment (yangc95) [reply content/javascript/concepts/setTimeout/setTimeout.md:32]: I'd like to discuss what potential headers could work better here. We'll keep this as an h2.

**Comment by yangc95** [comment]
> @yangc95 already made changes suggested, please check

@RaffyRod Awesome! Just did a check and looks like some comments were glossed over. I also left some comments that may require a bit more working on/discussion.

File: content/javascript/concepts/setTimeout/setTimeout.md
Comment (RaffyRod) [reply content/javascript/concepts/setTimeout/setTimeout.md]: @yangc95 ready, please check

**Comment by RaffyRod** [comment]
@yangc95 already made changes suggested, please check

**Comment by KTom101** [comment]
@RaffyRod It looks like a few more comments from @yangc95 need to be addressed.

**Comment by RaffyRod** [comment]
@KTom101 I already added the changes, now just waiting for another review from @yangc95

**Comment by KTom101** [comment]
@RaffyRod I think a few of the changes might have not gone through. When I click the "files changed" tab at the top (the image of the tab I'm clicking on is shown below) and scroll through the file I still see about 6 comments that haven't been resolved.

Screenshot 2023-05-13 at 9 33 26 PM

**Comment by RaffyRod** [comment]
@KTom101 Will check again , thanks for letting me know

**Comment by KTom101** [comment]
@RaffyRod No problem!

File: content/javascript/concepts/setTimeout/setTimeout.md
Comment (RaffyRod) [reply content/javascript/concepts/setTimeout/setTimeout.md]: Already commited some new titles, please check @yangc95

**Comment by RaffyRod** [comment]
@KTom101 please check now

**Comment by KTom101** [comment]
@RaffyRod Once @yangc95 approves the changes, I will conduct a second review. 👍

File: content/javascript/concepts/setTimeout/setTimeout.md
Comment (yangc95) [inline content/javascript/concepts/setTimeout/setTimeout.md:3]:
```suggestion
Description: 'Allows users to execute callbacks after a period of time expressed in milliseconds.'
```

File: content/javascript/concepts/setTimeout/setTimeout.md
Comment (yangc95) [inline content/javascript/concepts/setTimeout/setTimeout.md:16]:
```suggestion
**`setTimeout()`** is a function serviced globally by the window object provided by the user's browser.
```

File: content/javascript/concepts/setTimeout/setTimeout.md
Comment (yangc95) [inline content/javascript/concepts/setTimeout/setTimeout.md:22]:
```suggestion
The provided sample code contains some examples of how the function is composed:
```

File: content/javascript/concepts/setTimeout/setTimeout.md
Comment (yangc95) [inline content/javascript/concepts/setTimeout/setTimeout.md:52]:
```suggestion
## Example 1
```

File: content/javascript/concepts/setTimeout/setTimeout.md
Comment (yangc95) [inline content/javascript/concepts/setTimeout/setTimeout.md:69]:
```suggestion
## Codebyte Example
```

File: content/javascript/concepts/setTimeout/setTimeout.md
Comment (yangc95) [inline content/javascript/concepts/setTimeout/setTimeout.md:89]:
```suggestion
```

**Review by yangc95 — APPROVED** [review]
@RaffyRod To speed up this process, I went ahead and made some changes here: https://github.com/Codecademy/docs/pull/2114/commits/e0e91fc5a704eeacd4a877dd5dc491e0159d4651

Looks ready for a second review @KTom101

**Comment by yangc95** [comment]
Also making a note that this entry should be a term under window functions but I understand that the concept has yet to be created. Please refer to the original issue to see the file location.

Prior to merge, this should be moved to the terms folder under that concept.

cc: @KTom101

**Comment by RaffyRod** [comment]
@yangc95  @KTom101  I need help, I was pulling the latest changes from main branch and now mine says that I uploaded 48 changes, but I was working in just a file.

**Comment by KTom101** [comment]
@RaffyRod It looks like there was a force push to the branch in the commit history (see image below). I would try to avoid force pushing unless necessary. 

image

It may be possible to revert back to the previous commit. Do you remember what command you ran?

**Comment by RaffyRod** [comment]
> @RaffyRod It looks like there was a force push to the branch in the commit history (see image below). I would try to avoid force pushing unless necessary.
> 
> image
> 
> It may be possible to revert back to the previous commit. Do you remember what command you ran?

Don't remember to be honest, can you help me? otherwise, I can create a new PR with the latest changes for this

**Comment by RaffyRod** [comment]
@yangc95 please help..

**Comment by yangc95** [comment]
@RaffyRod Let's open a new PR then and stick to regular merges. Closing this one out but make sure to tag myself and @KTom101 with the new PR you submit.

Thanks!

**Pull request closed without merging**
